src/nolongerevil/integrations/mqtt/home_assistant_discovery.py

```diff
@@ -720,16 +594,17 @@ def get_all_discovery_configs(
         shared_values: Shared object values
         topic_prefix: MQTT topic prefix
         discovery_prefix: HA discovery prefix (default: homeassistant)
+        ha_mode: Current HA mode (heat_cool, heat, cool, off) for mode-aware discovery
```

Comment:
I sort of disagree with this idea, which uses the HA mode as part of discovery, when we already have the shared_values, from which we ought to be able to derive the expected HA mode instead of passing around the HA mode